crates/transaction/reveal/src/visitor.rs

```diff
@@ -0,0 +1,66 @@
+use strata_primitives::hash;
+use strata_state::{batch::SignedBatchCheckpoint, da_blob::PayloadCommitment};
+
+pub trait EnvelopeVisitor<'t> {
+    type Type;
+    fn visit_envelope_payload_chunks(chunks: impl Iterator<Item = &'t [u8]>) -> Self::Type;
+}
+// CheckpointVisitor
+pub trait CheckpointVisitor<'t>: EnvelopeVisitor<'t> {
+    fn visit_checkpoint(&self) -> Option<SignedBatchCheckpoint>;
+}
+
```

Comment:
This isn't how visitors work.  They update their internal state bookkeeping (meaning the `visit_` fns have to take `&mut self`).  This `visit_checkpoint` fn doesn't make sense since it isn't taking an argument of what's being visited.